article/app/controllers/LiveBlogController.scala

```diff
@@ -80,10 +81,14 @@ class LiveBlogController(
     Action.async { implicit request: Request[AnyContent] =>
       val filter = shouldFilter(filterKeyEvents)
       val range = getRange(lastUpdate)
+
       mapModel(path, range, filter) {
         case (blog: LiveBlogPage, _) if rendered.contains(false) => getJsonForFronts(blog)
-        case (blog: LiveBlogPage, blocks) if request.forceDCR    => Future.successful(renderGuuiJson(blog, blocks, filter))
-        case (blog: LiveBlogPage, _)                             => getJson(blog, range, isLivePage, filter)
+        case (blog: LiveBlogPage, blocks) if request.forceDCR && lastUpdate.isEmpty =>
+          Future.successful(renderGuuiJson(blog, blocks, filter))
+        case (blog: LiveBlogPage, blocks) =>
+          val blocksFlattened = blocks.requestedBodyBlocks.getOrElse(Map.empty[String, Seq[Block]]).flatMap(_._2).toSeq
```

Comment:
is this getting all the "requested body blocks" regardless of the key they were requested with and merging them in a list? Should we not know what the key is that we're looking for? I'm guessing it would be based on the lastUpdated parameter. Might be clearer if we extract this into a method and be more explicit about what we're looking for, otherwise we might risk picking up block we didn't ask for if someone ever changes the capi query to add other thing.

Reply:
we can also do this as a separate PR btw